### ISSUE TITLE
plone.app.dexterity can not be installed. (was: Package dependencies are not properly installed in Plone 5.0a3).

### ISSUE THREAD

**Issue by tisto** [plone/Products.CMFPlone#381]
Package dependencies declared in profiles/default/metadata.xml are ignored in Plone 5.0a3:

```
<?xml version="1.0"?>
<metadata>
  <version>1000</version>
  <dependencies>
    <dependency>profile-plone.app.dexterity:default</dependency>
  </dependencies>
</metadata>
```

Steps to reproduce:

1) Create package with mr.bob "$ bin/mrbob -O test.plone_addon bobtemplates:plone_addon"
2) Add plone.app.dexterity to install_requires in setup.py
3) Add profile-plone.app.dexterity:default to dependencies in profiles/default/metadata.xml
4) Run buildout, start Plone and install the created add-on product
5) plone.app.dexterity will not be installed (see prefs_install_products_form)

Pull request with test failure on travis:

https://github.com/plone/bobtemplates.plone/pull/46
https://travis-ci.org/plone/bobtemplates.plone/builds/51644101

**Comment by tisto** [comment]
When I declare collective.mailchimp instead of plone.app.dexterity, it works. Also installing plone.app.dexterity directly does not work:

1) Go to prefs_install_products_form
2) Click "Install plone.app.dexterity"
3) Page is reloaded, plone.app.dexterity is still not installed

Same is true for plone.app.contenttypes.

**Comment by tisto** [comment]
Seems like all packages listed here can not be installed: https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/factory.py#L23

**Comment by davisagli** [comment]
Actually I think the plone.app.dexterity profile was successfully run when the site was created -- after all, the Dexterity Content Types control panel shows up. But because it is listed in the list of non-installable profiles, the quick installer filters it out here - https://github.com/plone/Products.CMFQuickInstallerTool/blob/master/Products/CMFQuickInstallerTool/events.py#L60 - and does not record that the product was installed. That's why portal_quickinstaller.isProductInstalled('plone.app.dexterity') fails. I suspect this was done intentionally so that these products do not end up with the option to uninstall them in portal_quickinstaller.

While looking into this I found a related bug, which is that plone.app.dexterity was showing up in the list of available products in the add-ons control panel. That was happening because the control panel was not filtering out non-installable profiles, which I fixed here: 35bba15124f19e4fbf583f3c9af2b341301df820
